microsoft/testsuites/dpdk/common.py

```diff
@@ -17,6 +22,266 @@
 AZ_ROUTE_ALL_TRAFFIC = "0.0.0.0/0"
 
 
+# Attempt to clean up the DPDK package dependency mess
+# Make a Installer class that implements the common steps
+# for installing DPDK/rdma-core, either from source or the package manager.
+# This generic class will get implemented in DpdkTestpmd and RdmaCore.
+# This should help us cover the various installation cases in a nice way,
+# and allow us to only re-implement the bits we need for each project.
+class OsPackageDependencies:
+    # A class to reduce the isinstance() trees that are
+    # sprinkled everywhere.
+    # Caller provides a function to match an OS and
+    # the packages to install on that OS.
+    def __init__(
+        self,
+        os_matcher: Callable[[Posix], bool],
+        packages: Sequence[Union[str, Tool, Type[Tool]]],
+    ) -> None:
+        self.check_node_os = os_matcher
+        self.packages = packages
+
+
+class InstallDependencies:
```

Comment:
define it as a function.

Reply:
If a class host only one function, and no data used, it should be a function. Some languages like C#, Jave are purely object oriented, so they cannot use function. But Python uses mixed functions and oo. Avoid overusing classes, so the remaining classes are easier to be understood.

Reply:
I appreciate all of that, I agree with all of what you've said. But:
The os match -> package list _is_ data. I think it's cleaner to encapsulate these dependencies so I can pass an instance of this class to the constructor for the Dpdk___Installer later. This gives the reader all the info they need to know about what's happening.

I can make a couple of DependencyInstaller instances for source and package manager installs. In our __init__ path I just call `self._installer=DpdkGitInstaller(dependencies=dpdk_source_dependencies)`
or  `self._installer=DpdkTarInstaller(dependencies=dpdk_source_dependencies)`
or  `self._installer=DpdkPackageManagerInstaller(dependencies=dpdk_package_manager_dependencies)`
Afterwards, in the install function I just call:

`self._installer.run_installation()`

If you look at the code we're replacing, this is a much more organized way to handle things. If we need to replace this later, I will go back and fix it. For now, this is how I want to handle it.

Reply:
Also remember: this code is going to be used for both dpdk and rdma-core eventually. Keeping that organized and readable is going to be important, I'm doing this intentionally to make that future PR easier to read 😂

I am not exaggerating, I wrote the DPDK 32bit test and looked at the PR and said to myself "Chi is never going to approve this"

So I went back and starting fixing the installation path 😜